gammapy/image/maps.py

```diff
@@ -82,14 +82,28 @@ def read(cls, fobj, *args, **kwargs):
                 fobj = str(make_path(fobj))
             data = fits.getdata(fobj, *args, **kwargs)
             header = fits.getheader(fobj, *args, **kwargs)
-        wcs = WCS(header)
-        meta = header
 
+        image_hdu = fits.ImageHDU(data, header)
+        return cls.from_image_hdu(image_hdu)
+
+    @classmethod
+    def from_image_hdu(cls, image_hdu):
+        """
+        Read sky map from ImageHDU.
```

Comment:
Change "Read ..." to "Create ..." here. For me "read" sounds like it's reading from disk, which is not what this is doing any more ... there's already an in-memory `ImageHDU` object. `from_image_hdu` is an alternative constructor.